backends/bevy_picking_sprite/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+
+use std::cmp::Ordering;
+
+use bevy::ui::FocusPolicy;
+use bevy::{prelude::*, window::PrimaryWindow};
+use bevy_picking_core::backend::prelude::*;
+
+/// Commonly used imports for the [`bevy_picking_sprite`](crate) crate.
+pub mod prelude {
+    pub use crate::SpriteBackend;
+}
+
+/// Adds picking support for [`bevy_sprite`](bevy::sprite)
+#[derive(Clone)]
+pub struct SpriteBackend;
+impl PickingBackend for SpriteBackend {}
+impl Plugin for SpriteBackend {
+    fn build(&self, app: &mut App) {
+        app.add_system(sprite_picking.in_set(PickSet::Backend));
+    }
+}
+
+/// Checks if any sprite entities are under each pointer
+pub fn sprite_picking(
+    pointers: Query<(&PointerId, &PointerLocation)>,
+    cameras: Query<(Entity, &Camera)>,
+    windows: Query<(Entity, &Window)>,
+    primary_window: Query<Entity, With<PrimaryWindow>>,
+    images: Res<Assets<Image>>,
+    sprite_query: Query<(
+        Entity,
+        &Sprite,
+        &Handle<Image>,
+        &GlobalTransform,
+        &ComputedVisibility,
+        Option<&FocusPolicy>,
+    )>,
+    mut output: EventWriter<EntitiesUnderPointer>,
+) {
+    let mut sorted_sprites: Vec<_> = sprite_query.iter().collect();
+    sorted_sprites.sort_by(|a, b| {
+        (b.3.translation().z)
+            .partial_cmp(&a.3.translation().z)
+            .unwrap_or(Ordering::Equal)
+    });
+
+    for (pointer, location) in pointers.iter().filter_map(|(pointer, pointer_location)| {
+        pointer_location.location().map(|loc| (pointer, loc))
+    }) {
+        let cursor_position = location.position;
+        let mut blocked = false;
+
+        let camera = cameras
+            .iter()
+            .find(|(_entity, camera)| {
+                camera
+                    .target
+                    .normalize(Some(primary_window.single()))
+                    .unwrap()
+                    == location.target
+            })
+            .map(|(entity, _camera)| entity)
+            .unwrap_or_else(|| panic!("No camera found associated with pointer {:?}.", pointer));
+
+        let over_list = sorted_sprites
+            .iter()
+            .copied()
+            .filter_map(
+                |(entity, sprite, image, global_transform, visibility, focus)| {
+                    if blocked || !visibility.is_visible() {
+                        return None;
+                    }
+
+                    blocked = focus != Some(&FocusPolicy::Pass);
+
+                    let position = global_transform.translation();
+                    let sprite_position = position.truncate();
+
+                    let extents = sprite
+                        .custom_size
+                        .or_else(|| images.get(image).map(|f| f.size()))
+                        .map(|size| size / 2.0)?;
+
+                    let anchor_offset = sprite.anchor.as_vec() * extents;
+
+                    let target_size = if let Some(t) =
+                        location.target.get_render_target_info(&windows, &images)
```

Comment:
I think this is a "complicated" research of target size, half target size depending on where the Y is on the screen, not taking into account camera position... etc etc.

There is a bevy builtin way to get world position from a camera:
```rs
let Some(cursor_position) = camera
            .viewport_to_world(camera_transform, cursor_position)
            .and_then(|ray| Some(ray.get_point(0f32))) else {
                continue;
            };
```

Check it in "real code" there: https://github.com/Vrixyz/bevy_jam_3/blob/a58ce17b9f03b1ab5e36ac81acfa070576cc1d1f/src/picking_aabb.rs#L72 ; adapted from your sprite backend, hope you don't mind :)

Reply:
I don't think this is applicable for the sprite case. There is no way to get the sprite bounds in world coordinates that I am aware of, so I have to compute it in screen space.

Reply:
Oh the bounds of a sprite is expressed in screen size ? Alright, I'm just worried this solution won't support camera position, rotation and scale/zoom ; anyway it's more an implementation thing and could be adressed easily after this big PR if need be 👍

Reply:
> I'm just worried this solution won't support camera position, rotation and scale/zoom.

Ah! I misunderstood. I'm not very familiar with 2D, but it does seem like there should be a better way to convert a screen space coordinate to world space for 2D.

Reply:
This should now be fixed in 53e4dfe2696822742f187e12165ea06eaad690e6, thanks for raising this issue!